File: src/main/java/org/dbflute/intro/app/web/dfprop/settings/DfpropSettingsResult.java
/*
 * Copyright 2014-2021 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */
package org.dbflute.intro.app.web.dfprop.settings;

import javax.validation.Valid;

import org.dbflute.intro.dbflute.allcommon.CDef;
import org.lastaflute.core.util.Lato;
import org.lastaflute.web.validation.Required;

/**
 * @author hakiba
 * @author jflute
 */
public class DfpropSettingsResult {

    // ===================================================================================
    //                                                                           Attribute
    //                                                                           =========
    /** DBFluteクライアントのプロジェクト名 e.g. maihamadb */
    @Required
    public String projectName;

    /** DBMSを識別するコード e.g. mysql */
    @Required
    public CDef.TargetDatabase databaseCode;

    /** 自動生成コードの言語を識別するコード e.g. java */
    @Required
    public CDef.TargetLanguage languageCode;

    /** 自動生成コードが利用するDIコンテナを識別するコード e.g. lasta_di */
    @Required
    public CDef.TargetContainer containerCode;

    /** 自動生成コードの基底パッケージ (dbfluteはここでは含まない) e.g. "org.docksidestage.showbase" */
    @Required
    public String packageBase;

    /** JDBCドライバーのクラス名 e.g. "com.mysql.jdbc.Driver" */
    @Required
    public String jdbcDriverFqcn;

    /** DBのメインスキーマの接続情報、databaseInfoMap.dfpropに記載されているもの */
    @Required
    @Valid
    public DatabaseSettingsPart mainSchemaSettings;

    public static class DatabaseSettingsPart {

        /** JDBCの接続URL e.g. "jdbc:mysql://localhost:3306/maihamadb" */
        @Required
        public String url;

        /** JDBCの接続スキーマ、DBMSによっては指定なし e.g. maihamadb */
        public String schema;

        /** JDBCの接続ユーザー e.g. maihamauser */
        @Required
        public String user;

        /** JDBCの接続パスワード、パスワードなしなら空っぽ e.g. maihamapass */
        public String password;
    }

    /** DBFluteクライアントが利用するDBFluteのバージョン e.g. "1.2.6" */
    @Required
    public String dbfluteVersion;

    // TODO you 画面で使ってるところが見つからない... by jflute (2023/01/13)
    /** extlib配下のJDBCドライバーのjarファイルのcanonical path e.g. "/Users/intro/xxx.../extlib/yyy.jar" (NullAllowed) */
    public String jdbcDriverJarPath;

    // ===================================================================================
    //                                                                      Basic Override
    //                                                                      ==============
    @Override
    public String toString() {
        return Lato.string(this);
    }
}

File: frontend/src/static/app/api/intro/dfprop/settings/DfpropSettingsResult.ts
/*
 * Copyright 2014-2021 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

/**
 * The bean class as return for remote API of GET /dfprop/settings/{projectName}.
 * @author FreeGen
 */
type DfpropSettingsResult = {
  /** DBFluteクライアントのプロジェクト名 e.g. maihamadb (Required) */
  projectName: string

  /** DBMSを識別するコード e.g. mysql: * `mysql` - MySQL. * `postgresql` - PostgreSQL. * `oracle` - Oracle. * `db2` - Db2, DB2. * `sqlserver` - SQLServer. * `h2` - H2Database, H2 Database. * `derby` - ApacheDerby, Apache Derby. :: fromCls(CDef$TargetDatabase) (enumValue=[mysql, postgresql, oracle, db2, sqlserver, h2, derby]) (Required) */
  databaseCode: string

  /** 自動生成コードの言語を識別するコード e.g. java: * `java` - Java. * `csharp` - C, C#. * `scala` - Scala. :: fromCls(CDef$TargetLanguage) (enumValue=[java, csharp, scala]) (Required) */
  languageCode: string

  /** 自動生成コードが利用するDIコンテナを識別するコード e.g. lasta_di: * `lasta_di` - LastaDi, Lasta Di. * `spring` - SpringFramework, Spring Framework. * `guice` - GoogleGuice, Google Guice. * `seasar` - SeasarS2Container, Seasar (S2Container). * `cdi` - Cdi, CDI. :: fromCls(CDef$TargetContainer) (enumValue=[lasta_di, spring, guice, seasar, cdi]) (Required) */
  containerCode: string

  /** 自動生成コードの基底パッケージ (dbfluteはここでは含まない) e.g. "org (Required) */
  packageBase: string

  /** JDBCドライバーのクラス名 e.g. "com (Required) */
  jdbcDriverFqcn: string

  /** (Required) */
  mainSchemaSettings: DfpropSettingsResult_DatabaseSettingsPart

  /** DBFluteクライアントが利用するDBFluteのバージョン e.g. "1 (Required) */
  dbfluteVersion: string

  /** extlib配下のJDBCドライバーのjarファイルのcanonical path e.g. "/Users/intro/xxx (NullAllowed) */
  jdbcDriverJarPath?: string
}

/**
 * The part class of DatabaseSettingsPart.
 * @author FreeGen
 */
type DfpropSettingsResult_DatabaseSettingsPart = {
  /** JDBCの接続URL e.g. "jdbc:mysql://localhost:3306/maihamadb" (Required) */
  url: string

  /** JDBCの接続スキーマ、DBMSによっては指定なし e.g. maihamadb (NullAllowed) */
  schema?: string

  /** JDBCの接続ユーザー e.g. maihamauser (Required) */
  user: string

  /** JDBCの接続パスワード、パスワードなしなら空っぽ e.g. maihamapass (NullAllowed) */
  password?: string
}
